### ISSUE TITLE
TensorFlow Reddit Dataset to DataFrame Throws Error

### ISSUE THREAD

**Issue by RylanSchaeffer** [tensorflow/tensorflow#48592]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): source
- Python version: 3.6.9
- TensorFlow version (use command below):

```
>>> tf.version.GIT_VERSION
'v2.4.0-49-g85c8b2a817f'
>>> tf.version.VERSION
'2.4.1'
```

**Describe the current behavior**

I want to load the [tfds reddit dataset](https://www.tensorflow.org/datasets/catalog/reddit) and convert it to a dataframe. The below standalone code throws the following error:
```
  File "/home/rylan/Documents/FieteLab-RCRP/exp_03_language_modeling/main.py", line 36, in load_dataset
    reddit_dataframe = tfds.as_dataframe(reddit_dataset)
  File "/home/rylan/Documents/FieteLab-RCRP/rcrp/lib/python3.6/site-packages/tensorflow_datasets/core/as_dataframe.py", line 218, in as_dataframe
    df = StyledDataFrame(rows)
  File "/home/rylan/Documents/FieteLab-RCRP/rcrp/lib/python3.6/site-packages/tensorflow_datasets/core/as_dataframe.py", line 144, in __init__
    super().__init__(*args, **kwargs)
TypeError: object.__init__() takes no parameters
python-BaseException
```

**Standalone code to reproduce the issue**

```
import tensorflow as tf
import tensorflow_datasets as tfds


reddit_dataset = tfds.load(
    'reddit',
    split='train',
    shuffle_files=False,
    download=True,
    data_dir=data_dir)
assert isinstance(reddit_dataset, tf.data.Dataset)
reddit_dataframe = tfds.as_dataframe(reddit_dataset)

```

**Comment by RylanSchaeffer** [comment]
The error goes away and a sensible result is produced if I remove these two lines https://github.com/tensorflow/datasets/blob/master/tensorflow_datasets/core/as_dataframe.py#L222-L223 and add `df = pd.DataFrame(rows)`.

**Comment by RylanSchaeffer** [comment]
Ok upon further investigation, I found the cause of the error. I'm using Tensorflow Datasets 4.2.0, which has the following at the top of the `as_dataframe.py` file:

```
try:
  import pandas  # pylint: disable=g-import-not-at-top
  import pandas.io.formats.style  # pylint: disable=g-import-not-at-top
  DataFrame = pandas.DataFrame
except ImportError:
  DataFrame = object
```

The second import statement was failing, causing `DataFrame` to be set to `object`.  I noticed this import was removed in the most recent commit. 

https://github.com/tensorflow/datasets/commit/81e8ec79e9782f6eb5a93689bef9a5a076a3c2d4

For anyone else with a similar problem, the solution is to remove `import pandas.io.formats.style`.
